### PR TITLE
fix: Add gitpod configuration

### LINKED ISSUE #518: Correcting Gitpod Configuration: Updating gitpod.yml for Proper Port Exposure and Initialization Commands
Gitpod isn't configured properly, we need to configure the gitpod.yml file to expose the right ports and run the necessary commands to start the dev env.

image

### PR DESCRIPTION
**Pull request by anikdhabal**
Previously there was no gitpod.yml file to setup in gitpod. I've created the gitpod.yml and add required commands.

fixes #518 

result:-

Screenshot 2023-10-05 173104

Screenshot 2023-10-05 173215

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    addCheck: true
+    addComment: true
+    addBadge: true
+
+vscode:
+  extensions:
+    - bradlc.vscode-tailwindcss
+    - esbenp.prettier-vscode
+    - Prisma.prisma
+
```
Comment (anjy7) [inline .gitpod.yml:44]: - DavidAnson.vscode-markdownlint
    - yzhang.markdown-all-in-one
    - dbaeumer.vscode-eslint
    - ban.spellright

Comment (anjy7) [reply .gitpod.yml:44]: Just a tiny change - please add these extensions as well

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - port: 9000
+    visibility: public
+    onOpen: ignore
+  - port: 1100
+    visibility: private
+    onOpen: ignore
+  - port: 2500
+    visibility: private
+    onOpen: ignore
+
```
Comment (anjy7) [inline .gitpod.yml:28]: - port: 54320
    visibility: private
    onOpen: ignore

File: .gitpod.yml (hunk 1 above)
Comment (anikdhabal) [reply .gitpod.yml:44]: I don't think we need add these extension.

Comment (anikdhabal) [reply .gitpod.yml:44]: I don't think we need add these extension.

**Comment by anikdhabal** [comment]
cc @dguyen

File: .gitpod.yml (hunk 1 above)
Comment (Mythie) [reply .gitpod.yml:47]: Hey we have a comprehensive list of extensions at https://github.com/documenso/documenso/blob/feat/refresh/.devcontainer/devcontainer.json#L16

```json
[
  "aaron-bond.better-comments",
  "bradlc.vscode-tailwindcss",
  "dbaeumer.vscode-eslint",
  "esbenp.prettier-vscode",
  "mikestead.dotenv",
  "unifiedjs.vscode-mdx",
  "GitHub.copilot-chat",
  "GitHub.copilot-labs",
  "GitHub.copilot",
  "GitHub.vscode-pull-request-github",
  "Prisma.prisma",
  "VisualStudioExptTeam.vscodeintellicode"
]
```

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+tasks:
+  - init: |
+      npm i &&
+      npm run dx:up &&
+      cp .env.example .env &&
+      set -a; source .env &&
+      sed -i -e "s|http://localhost:3000|$(gp url 3000)|" .env \
+             -e "s|http://localhost:3001|$(gp url 3001)|" .env \
+             -e "s|http://localhost:9000|$(gp url 9000)|" .env
```
Comment (Mythie) [inline .gitpod.yml:9]: Just use `export NEXT_AUTH="$(gp url 3000)"` for these items rather than `sed` please

File: .gitpod.yml (hunk 1 above)
Comment (anikdhabal) [reply .gitpod.yml:47]: Ok let me add that.

File: .gitpod.yml (hunk 3 above)
Comment (anikdhabal) [reply .gitpod.yml:9]: okk

**Comment by anikdhabal** [comment]
cc @dguyen @Mythie

File: .gitpod.yml (hunk 3 above)
Comment (anikdhabal) [reply .gitpod.yml]: Done

**Comment by anikdhabal** [comment]
cc @dguyen @Mythie

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
Thanks Anik! 🎉 

thanks-pusheen

**Comment by anikdhabal** [comment]
> Thanks Anik! 🎉
> 
> thanks-pusheen [ thanks-pusheen ](https://user-images.githubusercontent.com/13398220/274156632-b2570036-2e87-4a12-b126-2c3be6d4925c.gif) [ ](https://user-images.githubusercontent.com/13398220/274156632-b2570036-2e87-4a12-b126-2c3be6d4925c.gif)

Happy to contribute🎉

**Pull request merged**